File: frontend/src/lib/api/repoLogs.ts
import { publicApiBaseUrl } from '$lib/api/config';

async function readJson<T>(res: Response): Promise<T> {
	if (!res.ok) {
		const text = await res.text();
		throw new Error(text || `HTTP ${res.status}`);
	}
	return (await res.json()) as T;
}

export type CommitLogSidebarEntry = {
	commit_sha: string;
	commit_message: string;
	committed_at: string;
	log_count: number;
};

export type CommitLogEntry = {
	action: string;
	spec_folder: string;
	message: string;
	reasoning: string;
	created_at: string;
};

export type RepoLogsSidebarResponse = {
	commits: CommitLogSidebarEntry[];
};

export type RepoCommitLogsResponse = {
	commit_sha: string;
	commit_message: string;
	committed_at: string;
	logs: CommitLogEntry[];
};

export async function fetchLogsSidebar(repoId: string | number): Promise<RepoLogsSidebarResponse> {
	const res = await fetch(`${publicApiBaseUrl()}/repos/${repoId}/logs/sidebar`);
	return readJson<RepoLogsSidebarResponse>(res);
}

export async function fetchCommitLogs(
	repoId: string | number,
	commitSha: string
): Promise<RepoCommitLogsResponse> {
	const url = new URL(`${publicApiBaseUrl()}/repos/${repoId}/logs`);
	url.searchParams.set('commit_sha', commitSha);
	const res = await fetch(url);
	return readJson<RepoCommitLogsResponse>(res);
}

File: api/src/specvsreality_api/schemas/repo_logs.py
"""Response schemas for commit decision logs."""

from __future__ import annotations

from datetime import datetime

from pydantic import BaseModel, ConfigDict


class CommitLogSidebarEntry(BaseModel):
    model_config = ConfigDict(extra="forbid")

    commit_sha: str
    commit_message: str
    committed_at: datetime
    log_count: int


class CommitLogEntry(BaseModel):
    model_config = ConfigDict(extra="forbid")

    action: str
    spec_folder: str
    message: str
    reasoning: str
    created_at: datetime


class RepoLogsSidebarResponse(BaseModel):
    model_config = ConfigDict(extra="forbid")

    commits: list[CommitLogSidebarEntry]


class RepoCommitLogsResponse(BaseModel):
    model_config = ConfigDict(extra="forbid")

    commit_sha: str
    commit_message: str
    committed_at: datetime
    logs: list[CommitLogEntry]
